xdsl/dialects/memref.py

```diff
@@ -371,6 +372,22 @@ def from_memref(memref: Operation | SSAValue):
         return Rank.build(operands=[memref], result_types=[IndexType()])
 
 
+@irdl_op_definition
+class CollapseShapeOp(IRDLOperation):
+    name = "memref.collapse_shape"
+    src: Operand = operand_def(MemRefType)
+    result: OpResult = result_def(MemRefType)
+    reassociation: AnyArrayAttr = prop_def(AnyArrayAttr)
```

Comment:
@superlopuh How can I restrict this to check only for i64 `ArrayAttr`s?

Reply:
I think the best we can do is this:
```suggestion
    reassociation = prop_def(ArrayAttr[IntegerAttr[IntegerType]])
```
And then still have a manual verify impl that checks for 64 bits

Reply:
I think you can replace IntegerAttr by Annotated[IntegerAttr, IntegerAttr(i64)]

Reply:
```suggestion
     reassociation = prop_def(ArrayAttr[Annotated[IntegerType, IntegerType(64)]])
```
Seems to be doing the trick! Thanks for the suggestions!

Reply:
Okay nope it didn't anyway, I think I fixed it now?